pkg/virt-operator/resource/generate/components/crds.go

```diff
@@ -91,6 +91,9 @@ func addFieldsToAllVersions(crd *extv1.CustomResourceDefinition, fields ...inter
 
 func patchValidation(crd *extv1.CustomResourceDefinition, version *extv1.CustomResourceDefinitionVersion) error {
 	name := crd.Spec.Names.Singular
+	if name == "" {
```

Comment:
@xpivarc could you add a test?

What was the case where you encountered this?

Reply:
This is the case https://github.com/kubevirt/kubevirt/pull/6026#discussion_r668521320 . I can have a look on what we can test

Reply:
so, this issue doesn't actually exist today then, right?  If not, maybe we should just fail the validation if no singular name is provided for the crd to ensure this doesn't occur. That would force the crd definitions to always have singular set or CI will fail.

Reply:
There isn't a issue. But you can have CRD without the `Singular`. In that case the lowercase `kind` is used... So I would say we can allow it by this PR.